crates/chat-cli/src/util/directories.rs

```diff
@@ -185,7 +185,44 @@ pub fn canonicalizes_path(os: &Os, path_as_str: &str) -> Result<String> {
     let context = |input: &str| Ok(os.env.get(input).ok());
     let home_dir = || os.env.home().map(|p| p.to_string_lossy().to_string());
 
-    Ok(shellexpand::full_with_context(path_as_str, home_dir, context)?.to_string())
+    let expanded = shellexpand::full_with_context(path_as_str, home_dir, context)?;
+    
+    // Convert all relative paths to absolute paths (including glob patterns)
+    if !expanded.starts_with("/") && !expanded.starts_with("~") {
+        let current_dir = os.env.current_dir()?;
+        let absolute_path = current_dir.join(expanded.as_ref());
+        // Normalize the path to remove ./ and ../ components
+        match absolute_path.canonicalize() {
```

Comment:
I am wondering should we just always use the normalize_path() method for code simplicity and consistency?